agent/config/config.go

```diff
@@ -373,20 +400,37 @@ func ec2MetadataConfig(ec2client ec2.EC2MetadataClient) Config {
 // error is returned, however, if the config is incomplete in some way that is
 // considered fatal.
 func NewConfig(ec2client ec2.EC2MetadataClient) (config *Config, err error) {
-	ctmp := environmentConfig() //Environment overrides all else
+	var errs []error
+	ctmp, env_err := environmentConfig() //Environment overrides all else
```

Comment:
You can reuse a single `innerErr` many times so you avoid needing to declare a bunch of separate ones.

Then in your `defer` you can assign the one that gets returned.  (Or, you could change the named return to `retErr` and just use `err` within the function body.)